Enforce minimum gcc compiler version

dev at 3b64ea39863999c50aa42013f33c12f0cb4d23da

`cmake ../src`

g++ 
`v4.9.3` fails[1]
`v6.1.0` succeeds
`v10.2.1` succeeds
 
Perhaps add minimum compiler version check

[1]
```
cd /home/rountree/copperopolis/variorum/cztb2_build/thirdparty_builtin/googletest/googletest && /usr/tce/packages/gcc/gcc-4.9.3/bin/c++  -DGTEST_HAS_TR1_TUPLE=0 -I/home/rountree/copperopolis/variorum/src/thirdparty_builtin/googletest/googletest/include -I/home/rountree/copperopolis/variorum/src/thirdparty_builtin/googletest/googletest  -std=c++11 -O2 -g -DNDEBUG -fPIC   -Wall -Wshadow -DGTEST_HAS_PTHREAD=1 -fexceptions -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -std=c++11 -o CMakeFiles/gtest.dir/src/gtest-all.cc.o -c /home/rountree/copperopolis/variorum/src/thirdparty_builtin/googletest/googletest/src/gtest-all.cc
In file included from /home/rountree/copperopolis/variorum/src/thirdparty_builtin/googletest/googletest/include/gtest/internal/gtest-death-test-internal.h:46:0,
                 from /home/rountree/copperopolis/variorum/src/thirdparty_builtin/googletest/googletest/include/gtest/gtest-death-test.h:43,
                 from /home/rountree/copperopolis/variorum/src/thirdparty_builtin/googletest/googletest/include/gtest/gtest.h:60,
                 from /home/rountree/copperopolis/variorum/src/thirdparty_builtin/googletest/googletest/src/gtest-all.cc:38:
/home/rountree/copperopolis/variorum/src/thirdparty_builtin/googletest/googletest/include/gtest/gtest-matchers.h: In static member function ‘static constexpr bool testing::internal::MatcherBase<T>::IsInlined()’:
/home/rountree/copperopolis/variorum/src/thirdparty_builtin/googletest/googletest/include/gtest/gtest-matchers.h:434:12: error: ‘is_trivially_copy_constructible’ is not a member of ‘std’
            std::is_trivially_copy_constructible<M>::value &&
            ^
/home/rountree/copperopolis/variorum/src/thirdparty_builtin/googletest/googletest/include/gtest/gtest-matchers.h:434:50: error: expected primary-expression before ‘>’ token
            std::is_trivially_copy_constructible<M>::value &&
                                                  ^
/home/rountree/copperopolis/variorum/src/thirdparty_builtin/googletest/googletest/include/gtest/gtest-matchers.h:434:51: error: ‘::value’ has not been declared
            std::is_trivially_copy_constructible<M>::value &&
```

--- a/src/CMake/Setup3rdParty.cmake
+++ b/src/CMake/Setup3rdParty.cmake
@@ -33,6 +33,14 @@ if(BUILD_TESTS)
     # Enable GTest
     ################################
 
+    # g++ v4.9.3 no longer compiles googletest.
+    # g++ v6.1.0 and later are known to work.
+    if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+        if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 6.1)
+            message(FATAL_ERROR "Minimum GNU g++ version is 6.1.0, found ${CMAKE_CXX_COMPILER_VERSION}")
+        endif()
+    endif()
+
     #
     # We always want to build gtest as a static lib, however
     # it shares our "BUILD_SHARED_LIBS" option, so we need

